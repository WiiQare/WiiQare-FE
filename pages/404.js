import Link from "next/link";
import React from "react";
import { BiLeftArrow } from "react-icons/bi";
import Image from "next/image";

const Error404 = () => {
  return (
    <>
      <div className="min-w-screen min-h-screen bg-blue-100 flex items-center p-5 lg:p-20 overflow-hidden relative">
        <div className="flex-1 min-h-full min-w-full rounded-3xl bg-white shadow-xl p-10 lg:p-20 text-gray-800 relative md:flex items-center text-center md:text-left">
          <div className="w-full md:w-1/2">
            <div className="mb-10 lg:mb-20">
              <Image
                alt="logo"
                src="/images/logo_dark.png"
                width={40}
                height={40}
                className="h-10"
              />
            </div>
            <div className="mb-10 md:mb-20 text-gray-600 font-light">
              <h1 className="font-black uppercase text-3xl lg:text-5xl text-orange mb-10">
                You seem to be lost!
              </h1>
              <p>The page you&apos;re looking for isn&apos;t available.</p>
              <p>Try searching again or use the Go Back button below.</p>
            </div>
            <div className="mb-20 md:mb-0">
              <Link href="/" legacyBehavior>
                <a className="text-lg font-light outline-none focus:outline-none transform transition-all hover:scale-110 text-orange hover:text-yellow-600 flex items-center gap-1">
                  <BiLeftArrow /> Go Back
                </a>
              </Link>
            </div>
          </div>
          <div className="w-full md:w-1/2 text-center">
            <Image alt="404" src="/images/404.png" height={40} width={40} />
            <a
              href="https://www.freepik.com/vectors/business"
              target="_blank"
              className="text-xs text-gray-300"
            >
              WiiQare Ã  votre service
            </a>
          </div>
        </div>
        <div className="w-64 md:w-96 h-96 md:h-full bg-blue-200 bg-opacity-30 absolute -top-64 md:-top-96 right-20 md:right-32 rounded-full pointer-events-none -rotate-45 transform"></div>
        <div className="w-96 h-full bg-yellow-200 bg-opacity-20 absolute -bottom-96 right-64 rounded-full pointer-events-none -rotate-45 transform"></div>
      </div>

      <div className="flex items-end justify-end fixed bottom-0 right-0 mb-4 mr-4 z-10">
        <div>
          <a
            title="Buy me a beer"
            href="#"
            target="_blank"
            className="bg-white flex items-center justify-center w-16 h-16 rounded-full transition-all shadow hover:shadow-lg transform hover:scale-110 hover:rotate-12"
          >
            <Image
              alt="favicon"
              className="object-cover object-center w-full h-full rounded-full"
              src="/images/favicon.png"
              width={40}
              height={40}
            />
          </a>
        </div>
      </div>
    </>
  );
};

export default Error404;
