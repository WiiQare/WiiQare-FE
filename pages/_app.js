import Head from "next/head";
import { SessionProvider } from "next-auth/react";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../theme";

import "../styles/globals.css";
import "../styles/main.css";

export default function App({ Component, pageProps }) {
  const getLayout = Component.getLayout ?? ((page) => page);
  return (
    <>
      <Head>
        <meta
          name="description"
          content="Payé les soins à vos familles depuis l'étranger"
        />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
        <script src="https://unpkg.com/flowbite@1.5.3/dist/flowbite.js"></script>

      </Head>
      <SessionProvider session={pageProps.session}>
        <ThemeProvider theme={theme}>        
          {getLayout(<Component {...pageProps} />)}
        </ThemeProvider>
      </SessionProvider>
    </>
  );
}
