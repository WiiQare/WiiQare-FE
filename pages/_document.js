import { Html, Head, Main, NextScript } from 'next/document';
import Drawer from '../components/molecules/Navbar/Drawer';

export default function Document() {
  return (
    <Html lang="en">
      <Head>
        <link rel="manifest" href="/manifest.json" />
        <link rel="apple-touch-icon" href="/images/favicon-192x192.png"></link>
        <meta name="theme-color" content="#fff" />
      </Head>
      <script src="https://unpkg.com/flowbite@1.5.3/dist/flowbite.js"></script>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}
