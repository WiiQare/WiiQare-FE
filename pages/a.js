import React, { useEffect } from 'react';
import { useSession } from "next-auth/react";
import Router from "next/router";

const A = () => {
    const {status, data} = useSession();

    useEffect(() => {
       if(status === 'unauthenticated') Router.replace('/login')
    }, [status]);

    if (status === 'authenticated') 
        return (
            <div>
                Protected
            </div>
        )

    return <>Loading...</>
}

export default A;
