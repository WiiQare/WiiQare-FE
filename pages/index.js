import Head from 'next/head'
import { getSession, useSession, signOut } from "next-auth/react"

export default function Home() {

	const { data:session } = useSession()
	
	const handleSignOut = () => {
		signOut()
	}

	return (
		<>
			<Head>
				<title>JAMII - UNICEF CRYPTO</title>
				<meta name="description" content="Payé les soins à vos familles depuis l'étranger" />
				<meta name="viewport" content="width=device-width, initial-scale=1" />
				<link rel="icon" href="/favicon.ico" />
			</Head>
			

			{session ? <User session={session} handleSignOut={handleSignOut} /> : <Guest />}
		</>
	)
}

function User({session, handleSignOut}) {
	return (
		<main>
			Homepage
		</main>
	)
}

function Guest() {
	return (
		<main className='container mx-auto text-center py-20 space-y-8'>
			<h3 className="text-4xl font-bold">Guest User</h3>
		</main>
	)
}

export async function getServerSideProps({req}) {
	const session = await getSession({req})

	if(!session) {
		return {
			redirect: {
				destination: '/login',
				permanent: false
			}
		}
	}

	return {props: {session}}
}