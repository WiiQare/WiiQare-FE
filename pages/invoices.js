import Head from "next/head";
import { getSession } from "next-auth/react";
import DashboardLayout from "../layouts/Dashboard";

const Page = () => {
  return (
    <>
      <Head>
        <title>Invoices</title>
      </Head>
      <div>
        <h2 className="text-black font-w700 mt-4">Invoices</h2>
      </div>
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;

export async function getServerSideProps({ req }) {
  const session = await getSession({ req });

  if (!session) {
    return {
      redirect: {
        destination: "/login",
        permanent: false,
      },
    };
  }

  return { props: { session } };
}
