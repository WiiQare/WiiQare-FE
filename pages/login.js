import Logo from "../components/atoms/LogoHeader";
import WelcomeSlider from "../components/molecules/WelcomeSlider";
import LoginForm from "../components/organisms/Auth/LoginForm";
import { LOGIN_SLIDES } from "../utils/constants";
import { useMediaQuery, CssBaseline } from "@mui/material";
import Head from "next/head";
import React from "react";

function Login() {
  const matches = useMediaQuery("(max-width: 992px)");
  return (
    <>
      <Head>
        <title>Wiiqare | Login</title>
        <link
          rel="stylesheet"
          href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.2/dist/css/bootstrap.min.css"
          integrity="sha384-xOolHFLEh07PJGoPkLv1IbcEPTNtaed2xpHsD9ESMhqIYd0nLMwNLD69Npy4HI+N"
          crossOrigin="anonymous"
        />
      </Head>
      <CssBaseline />
      <div id="form-section" className="container-fluid signin">
        <Logo />
        <div className="row">
          {!matches && <WelcomeSlider slides={LOGIN_SLIDES} />}
          <LoginForm />
        </div>
      </div>
    </>
  );
}

export default Login;
