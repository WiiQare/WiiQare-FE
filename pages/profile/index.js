import Head from "next/head";
import { getSession } from "next-auth/react";
import DashboardLayout from "../../layouts/Dashboard";
import Profile from "../../components/organisms/Profile";

const Page = () => {
  return (
    <>
      <Head>
        <title>My Profile</title>
      </Head>
        <Profile />
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;

export async function getServerSideProps({ req }) {
  const session = await getSession({ req });

  if (session) {
    return {
      redirect: {
        destination: "/login",
        permanent: false,
      },
    };
  }

  return { props: { session } };
}
