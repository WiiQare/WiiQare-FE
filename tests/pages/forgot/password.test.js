import Page from '@/pages/forgot/password';
import { render } from '@testing-library/react';
import { DrawContext } from '@/pages/_app';
import { SessionProvider } from 'next-auth/react';
import { QueryClientProvider, QueryClient } from 'react-query';
require('jest-fetch-mock').enableMocks();

fetch.mockResponse('[]');

describe('Forgot Password', () => {
  it('should render the page', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <SessionProvider session={{ user: { data: { userId: 'random123' } } }}>
          <DrawContext.Provider
            value={{
              draw: false,
              setDraw: jest.fn(),
              hideSide: false,
              setHideSide: jest.fn(),
            }}
          >
            <Page />
          </DrawContext.Provider>
        </SessionProvider>
      </QueryClientProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
