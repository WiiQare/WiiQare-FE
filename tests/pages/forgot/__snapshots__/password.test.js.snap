// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password should render the page 1`] = `
<div>
  <div
    class="container-fluid signin"
    id="form-section"
  >
    <div
      class="website-logo"
    >
      <a
        href="/"
      >
        <div
          class="logo"
          style="width: 150px; height: 50px;"
        />
      </a>
    </div>
    <div
      class="row"
    >
      <div
        class="info-slider-holder"
      >
        <div
          class="info-holder"
        >
          <div
            class="mini-testimonials-slider"
          >
            <div
              class="slick-slider slick-initialized"
              dir="ltr"
            >
              <div
                class="slick-list"
              >
                <div
                  class="slick-track"
                  style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                >
                  <div
                    aria-hidden="false"
                    class="slick-slide slick-active slick-current"
                    data-index="0"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="details-holder"
                        >
                          <img
                            alt=""
                            class="photo"
                            data-nimg="1"
                            decoding="async"
                            height="1024"
                            loading="lazy"
                            src="/_next/image?url=%2Fimages%2Fhospital-bed-.png&w=3840&q=75"
                            srcset="/_next/image?url=%2Fimages%2Fhospital-bed-.png&w=1200&q=75 1x, /_next/image?url=%2Fimages%2Fhospital-bed-.png&w=3840&q=75 2x"
                            style="color: transparent;"
                            width="1096"
                          />
                          <div
                            class="bold-title"
                          >
                            L’accès aux soins de santé est l’un des plus grands défis en 
                            <span>
                              Afrique.
                            </span>
                          </div>
                          <h5>
                            WiiQare vous permet de garder de l'argent de côté au cas où vous seriez malade.
                          </h5>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="1"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="details-holder"
                        >
                          <img
                            alt=""
                            class="photo"
                            data-nimg="1"
                            decoding="async"
                            height="1024"
                            loading="lazy"
                            src="/_next/image?url=%2Fimages%2FWiiQare-is-secured.png&w=3840&q=75"
                            srcset="/_next/image?url=%2Fimages%2FWiiQare-is-secured.png&w=1200&q=75 1x, /_next/image?url=%2Fimages%2FWiiQare-is-secured.png&w=3840&q=75 2x"
                            style="color: transparent;"
                            width="1096"
                          />
                          <div
                            class="bold-title"
                          >
                            Facile à créer un compte 
                            <span>
                              sécurisé
                            </span>
                          </div>
                          <h5>
                            Créer un compte est simple et direct.
                          </h5>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="2"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="details-holder"
                        >
                          <img
                            alt=""
                            class="photo"
                            data-nimg="1"
                            decoding="async"
                            height="1024"
                            loading="lazy"
                            src="/_next/image?url=%2Fimages%2Fset-your-savings-%20.png&w=3840&q=75"
                            srcset="/_next/image?url=%2Fimages%2Fset-your-savings-%20.png&w=1200&q=75 1x, /_next/image?url=%2Fimages%2Fset-your-savings-%20.png&w=3840&q=75 2x"
                            style="color: transparent;"
                            width="1096"
                          />
                          <div
                            class="bold-title"
                          >
                            Définissez votre objectif d’épargne et mettez de l’argent de côté pour votre 
                            <span>
                               santé
                            </span>
                          </div>
                          <h5>
                            Décidez combien vous souhaitez épargner pour la santé et nous vous aiderons à atteindre votre objectif.
                          </h5>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    class="slick-slide"
                    data-index="3"
                    style="outline: none; width: 0px;"
                    tabindex="-1"
                  >
                    <div>
                      <div
                        style="width: 100%; display: inline-block;"
                        tabindex="-1"
                      >
                        <div
                          class="details-holder"
                        >
                          <img
                            alt=""
                            class="photo"
                            data-nimg="1"
                            decoding="async"
                            height="1024"
                            loading="lazy"
                            src="/_next/image?url=%2Fimages%2FEasy-payment-scan-QR-code.png&w=3840&q=75"
                            srcset="/_next/image?url=%2Fimages%2FEasy-payment-scan-QR-code.png&w=1200&q=75 1x, /_next/image?url=%2Fimages%2FEasy-payment-scan-QR-code.png&w=3840&q=75 2x"
                            style="color: transparent;"
                            width="1096"
                          />
                          <div
                            class="bold-title"
                          >
                            Faire un paiement est un ABC 
                            <span>
                              simple
                            </span>
                          </div>
                          <h5>
                            Pour effectuer un paiement, scannez simplement un code QR.
                          </h5>
                          <div
                            class="form-button flex flex-col gap-3"
                          >
                            <button
                              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
                              id="submit"
                              tabindex="0"
                              type="submit"
                            >
                              Commencez Maintenant
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                            <button
                              aria-label="Install app"
                              class="link-button bg-gray-800 py-2.5 text-md font-medium uppercase text-center text-white rounded-lg"
                              id="setup_button"
                              title="Install app"
                            >
                              Install WiiQare App
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                class="slick-dots"
                style="display: block;"
              >
                <li
                  class="slick-active"
                >
                  <button>
                    1
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    2
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    3
                  </button>
                </li>
                <li
                  class=""
                >
                  <button>
                    4
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="form-holder"
      >
        <div
          class="menu-holder"
        >
          <ul
            class="main-links"
          >
            <li>
              <details
                class="dropdown"
              >
                <summary
                  class="m-1 btn px-2 -py-1 bg-transparent text-gray-800 text-xs border-2 hover:border-gray-300 border-none space-x-1 hover:!bg-gray-100"
                >
                  <svg
                    aria-hidden="true"
                    class="text-gray-600"
                    fill="none"
                    height="23"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="23"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 5h12M9 3v2m1.048 9.5A18.022 18.022 0 016.412 9m6.088 9h7M11 21l5-10 5 10M12.751 5C11.783 10.77 8.07 15.61 3 18.129"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                   
                  <svg
                    aria-hidden="true"
                    class="text-gray-400"
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </summary>
                <ul
                  class="shadow-sm menu dropdown-content z-[1] w-max rounded-xl"
                >
                  <li>
                    <button>
                      Français
                    </button>
                  </li>
                  <li>
                    <button>
                      Anglais  
                    </button>
                  </li>
                </ul>
              </details>
            </li>
            <li>
              <div
                class="MuiBox-root css-1at62qq"
              >
                <a
                  class="normal-link"
                  href="/register"
                >
                  Don't you have an account ?
                </a>
              </div>
            </li>
            <li>
              <div
                class="MuiBox-root css-0"
              >
                <a
                  href="/register"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium bg-yellow !-ml-10 px-2 md:!ml-10 md:!px-5 css-sghohy-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    S'INSCRIRE
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="signin-signup-form"
        >
          <div
            class="flex flex-col md:w-3/4 w-full border-opacity-50 gap-4"
          >
            <div
              class="form-items !w-full !max-w-full flex flex-col items-center mx-auto"
            >
              <div
                class="md:w-3/4 mx-auto"
              >
                <div
                  class="form-title"
                >
                  Mot de passe oublié
                </div>
                <div
                  class="MuiBox-root css-19bt0nb"
                />
                <form
                  class=""
                  id="signinform"
                >
                  <div
                    class="css-1p5q5e5-MuiStack-root"
                  >
                    <div
                      class="space-y-1"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="outlined-basic"
                          id="outlined-basic-label"
                        >
                          Adresse email
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                            id="outlined-basic"
                            name="email"
                            type="text"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-yjsfm1"
                            >
                              <span>
                                Adresse email
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge bg-yellow css-jh47zj-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="submit"
                      >
                        Envoyer
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
